calico/openstack/t_etcd.py

```diff
@@ -527,6 +530,63 @@ def stop(self):
         self.elector.stop()
 
 
+class CalicoEtcdWatcher(EtcdWatcher):
+    """
+    An EtcdWatcher that watches our status-reporting subtree.
+
+    Responsible for parsing the events and passing the updates to the
+    mechanism driver.
+
+    We deliberately do not share an etcd client with the transport.
+    The reason is that, if we share a client then managing the lifecycle
+    of the client becomes an awkward shared responsibility (complicated
+    by the EtcdClusterIdChanged exception, which is only thrown once).
+    """
+
+    def __init__(self, calico_driver):
+        host = cfg.CONF.calico.etcd_host
+        port = cfg.CONF.calico.etcd_port
+        LOG.info("CalicoEtcdWatcher created for %s:%s", host, port)
+        super(CalicoEtcdWatcher, self).__init__(
+            "%s:%s" % (host, port),
+            FELIX_STATUS_DIR
+        )
+        self.calico_driver = calico_driver
+
+        # Register for felix uptime updates.
+        self.register_path(FELIX_STATUS_DIR + "/<hostname>/status",
```

Comment:
`<foo>` is a capture, the path dispatcher captures the path segment there and passes it to the registered function as the `hostname=...` kwarg.